### PR TITLE
[Doc] Clarify that session can also mean a ray cluster

### LINKED ISSUE #35093: [Glossary] The definition of "session" is not complete
### Description

Here is the current definition: https://docs.ray.io/en/latest/ray-references/glossary.html?highlight=session#term-Session

However, the /tm/ray/session_*/ structure indicates that a new session is created whenever a new ray cluster/runtime is started. People will find them in places like https://docs.ray.io/en/latest/ray-observability/ray-logging.html#logging-directory-structure

### Link

https://docs.ray.io/en/latest/ray-references/glossary.html?highlight=session#term-Session

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Clarify in the glossary that session can also mean a ray cluster

## Related issue number
Closes #35093

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Can we use bullets points to clearly list the different meanings? Something like: 

- A Ray Train/Tune session, ....
- A Ray Cluster instance,...


It's a bit hard for me to digest it when there are several paragraphs.

**Review by scottsun94 — APPROVED** [review]
LGTM. Thanks!

File: doc/source/ray-references/glossary.rst [hunk 1]
```diff
@@ -533,16 +533,19 @@ documentation, sorted alphabetically.
...
+
+          The session allows access to metadata such as which trial is being run,
+          information about the total number of workers as well as the rank of the
+          current worker. The session is also the interface through which an individual
+          Trainable can interact with the Tune experiment as a whole. This includes uses
+          such as reporting an individual trial’s metrics, saving/loading checkpoints,
+          and retrieving the corresponding dataset shards for each Train worker.
+
+        - A Ray cluster: in some cases the session also means a :term:`Ray Cluster`.
+          For example logs of a Ray cluster are stored under ``session_xxx/logs/``.
```
Comment (angelinalg) [inline doc/source/ray-references/glossary.rst:548]:
```suggestion
          For example, logs of a Ray cluster are stored under ``session_xxx/logs/``.
```

File: doc/source/ray-references/glossary.rst [hunk 2]
```diff
@@ -533,16 +533,19 @@ documentation, sorted alphabetically.
...
-        information about the total number of workers as well as the rank of the
-        current worker. The session is also the interface through which an individual
-        Trainable can interact with the Tune experiment as a whole. This includes uses
-        such as reporting an individual trial’s metrics, saving/loading checkpoints,
-        and retrieving the corresponding dataset shards for each Train worker.
+        - A Ray Train/Tune session: Tune session at the experiment execution layer
+          and Train session at the Data Parallel training layer
+          if running data-parallel distributed training with Ray Train.
+
+          The session allows access to metadata such as which trial is being run,
```
Comment (angelinalg) [inline doc/source/ray-references/glossary.rst:540]:
```suggestion
          The session allows access to metadata, such as which trial is being run,
```

File: doc/source/ray-references/glossary.rst [hunk 3]
```diff
@@ -533,16 +533,19 @@ documentation, sorted alphabetically.
...
+          information about the total number of workers as well as the rank of the
```
Comment (angelinalg) [inline doc/source/ray-references/glossary.rst:541]:
```suggestion
          information about the total number of workers, as well as the rank of the
```

**Review by angelinalg — APPROVED** [review]

**Pull request merged**